### PR TITLE
LPD-21217 Implement disabling sorting control & test

### PR DESCRIPTION
**Pull request by kresimir-coko**
https://liferay.atlassian.net/browse/LPD-21217

## Steps for local reproduction:
1. Ensure `feature.flag.LPS-186871=true`
2. Create an FDS with `/data-set-manager/fields` endpoint and use `FDSField` schema (screenshot 1)
3. Select the only parent field (`fdsViewFDSFieldRelationship.*`) (screenshot 2)
4. Ensure it's sorting capability is disabled (screenshot 3)

## Screenshot 1
screenshot 1

## Screenshot 2
screenshot 2

## Screenshot 3
screenshot 3

### CODE REVIEW

**Comment by kresimir-coko** [comment]
ci:test:playwright

**Comment by kresimir-coko** [comment]
ci:test:playwright

**Comment by kresimir-coko** [comment]
ci:test:playwright

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/api.ts [hunk 1]
```diff
@@ -147,3 +153,9 @@ export async function getAllPicklists(
 
 	return items;
 }
+
+export function isSortable(field: IFDSField | IField): boolean {
+	return !(
+		field.type === EFieldType.OBJECT || field.type === EFieldType.ARRAY
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/api.ts:159]: Handling array type of fields is technically not in the scope of the task. Based on @dsanz [research](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/2739274471/LPD-17838+-+Reseach+sorting+capabilities+for+leaf+elements), there are some array fields where sorting works, e.g. `cmultiselectpicklist`. Do we want to disable them all?

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/table/Table.tsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ import React, {useEffect, useState} from 'react';
 
 import {IFDSViewSectionProps} from '../../../FDSView';
 import {FDSViewType} from '../../../FDSViews';
-import {getFields} from '../../../api';
+import {getFields, isSortable} from '../../../api';
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/table/Table.tsx:26]: Is `isSortable` just imported in this one file? If yes, we can put it here and not export it.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/table/Table.tsx [hunk 3]
```diff
@@ -159,13 +159,18 @@ const SaveFDSFieldsModalContent = ({
 	const saveFDSFields = async () => {
 		setSaveButtonDisabled(true);
 
-		const creationData: Array<{name: string; type: string}> = [];
+		const creationData: Array<{
+			name: string;
+			sortable: boolean;
+			type: string;
+		}> = [];
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/table/Table.tsx:166]: The code in `SaveFDSFieldsModalContent` seems good, but we're still missing the same code in `AddFieldsModalContent`. It's 
```js
if (Liferay.FeatureFlags['LPS-186871']) {
    // AddFieldsModalContent
}
else {
    // SaveFDSFieldsModalContent
}
``` 
`Add*` implements treeview selection, `Save*` is the old list selection. The naming is a bit awkward, but :man_shrugging: We'll clean it up when we remove FF.

I believe the code as it is in this PR sets all fields as not sortable by default.

File: modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts [hunk 4]
```diff
@@ -75,6 +75,28 @@ test.describe('Add fields to a view and show them in a fragment', () => {
 			await expect(page.getByText('creator.id').first()).toBeVisible();
 			await expect(page.getByText('creator.name').first()).toBeVisible();
 		});
+
+		await test.step('Open modal to edit a field', async () => {
+			const dropdownMenuButton = page
+				.locator('.actions-cell button')
+				.first();
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-views-web/fields.spec.ts:82]: How do you know what field is first? Isn't `dateCreated` first, that should have sorting option enabled? I am more and more certain that we should avoid using `first()` whenever possible.

**Comment by markocikos** [comment]
Followup in https://github.com/liferay-frontend/liferay-portal/pull/4108

**Pull request closed without merging**